core/subscription/cache.go

```diff
@@ -86,13 +88,18 @@ func (e *entry) Get(key string) *types.Value {
 
 type cache struct {
 	entries map[string]*entry
+	sync.Mutex
```

Comment:
should we use the rwmutex and use readlock for get?

Reply:
you could just lock the whole spaceSubscription in `collectionObserver.fetchEntries`, the 99% of time the cache is accessed in one goroutine, and it's accessed a lot, so constant locking-unlocking will add performance overhead 

btw use `feature/chat` as target branch